src/clients/googleClassroom.js

```diff
@@ -8,3 +13,90 @@ export function createShareToClassroomUrl(snapshotKey, title) {
   uri.search = `snapshot=${snapshotKey}`;
   return BASE_URL + qs.stringify({url: uri.href, title});
 }
+
+export function createSnapshotUrl(assignemtnKey) {
+  const uri = document.createElement('a');
+  uri.href = '/';
+  uri.search = `snapshot=${assignemtnKey}`;
+  return uri.href;
+}
+
+export async function getCourses(accessToken) {
+  const gapi = await initGapi();
+  const client = await initGapiClient(gapi);
+  const classroom = await initGapiClientClassroom(client, accessToken);
+  const courses = await classroom.courses.list();
+  return courses.result.courses;
+}
+
+export async function createClassroomCourseWork(
+  accessToken,
+  workType,
+  courseId,
+  url,
+  title,
+) {
+  const gapi = await initGapi();
+  const client = await initGapiClient(gapi);
+  const classroom = await initGapiClientClassroom(client, accessToken);
+  const newCoursework = await classroom.courses.courseWork.create({
+    courseId,
+    title,
+    workType,
+    state: 'PUBLISHED',
+    dueDate: {
+      year: 2018,
+      month: 3,
+      day: 28,
+    },
+    dueTime: {
+      hours: 23,
+      minutes: 59,
+      seconds: 0,
+      nanos: 0,
+    },
+    materials: {
+      link: {
+        url,
+        title,
+        thumbnailUrl: 'N/A',
+      },
+    },
+  });
+  return newCoursework.result;
+}
+
+export async function submitClassroomAssignment(
+  accessToken,
+  courseId,
+  courseWorkId,
+  snapshotUrl,
+) {
+  const gapi = await initGapi();
+  const client = await initGapiClient(gapi);
+  const classroom = await initGapiClientClassroom(client, accessToken);
+  const submissions =
```

Comment:
I think the `studentSubmissions[0]` thing would look clearer/more intentional if destructured here.